Orange/widgets/unsupervised/owpca.py

```diff
@@ -192,6 +193,9 @@ def fit(self):
         self.start_button.setEnabled(False)
         if self.data is None:
             return
+        if self.data.is_sparse():
```

Comment:
It was placed here since `self.data is None` check is here and this is somehow related. Also all other checks, like `len(data.domain.attributes) == 0` and `len(data) == 0` are here.

Though I think this fits better into `set_data`, I put it here to be consistent with the current state of the code. I can move it into `set_data` if you think that will be more clear.